lib/localenv/tarballenv.go

```diff
@@ -0,0 +1,80 @@
+package localenv
+
+import (
+	"io"
+	"path/filepath"
+
+	libapp "github.com/gravitational/gravity/lib/app"
+	"github.com/gravitational/gravity/lib/pack"
+	"github.com/gravitational/gravity/lib/pack/encryptedpack"
+	"github.com/gravitational/gravity/lib/utils"
+
+	"github.com/gravitational/license"
+	"github.com/gravitational/trace"
+)
+
+// NewTarballEnvironment creates new environment with specified configuration
```

Comment:
I would say something along the lines of "creates a new local environment for the specified unpacked cluster image tarball".